### PR TITLE
Adding local run to new-project-with-sdk branch

### PR DESCRIPTION
**Pull request by filmaj**
This uses the new [deno-slack-runtime](https://github.com/slackapi/deno-slack-runtime) project to handle the `run` hook implementation expected by the Slack CLI.

Tested this using the latest build of the Slack CLI by:

1. Creating a project: `slack create -t ./path/to/this/repo/with/this/branch/checked/out`
2. `cd friendly-camel-whatever`
3. I have a dev Slack instance up (dev1788), but it doesn't use legit SSL certificates, so I had to edit the `run` hook inside `./.slack/slack.json` to add the "ignore SSL errors" flag to `deno`, so it looked something like: `deno run -q --unstable --allow-write --allow-read --allow-net --unsafely-ignore-certificate-errors=dev1788.slack.com https://deno.land/x/deno_slack_runtime@0.0.2/mod.ts`
4. Local run the thing (extra flags needed for dev instances): `slack run --apihost https://dev1788.slack.com --runtime deno1.19`
5. Try it out and see your functions get run locally!

### CODE REVIEW

**Comment by shapirone** [comment]
**TLDR: Updating function logic works perfectly, updating manifest file does not**

Did these steps for setup and starting the process:
1.  Run `hermes-dev create -t slackapi/deno-reverse-string -b local-run-npws`
2. Start process via `hermes-dev run --slackapi`
3. Open configured workspace in dev instance

This worked beautifully :chefs-kiss: for the following steps:

1. Run default reverse function
2. Modify function (do a `.sort()` instead of a `.reverse()`)
3. Run function again in client
4. --> See sorted string instead of reversed string

No new process or nothing!  However, these steps required restarting the process:

1. Run default reverse function
2. Modify manifest to add new parameter
3. Run function again in client
4. --> No new parameter in Shortcut until after restarting the process

After the new parameter was added though I was able to do the following without restarting the process again:
1. Import the Slack API client into my function
2. Change my function inputs to accept `token`
3. Instantiate the API client and Send a message
4. ---> Slack message succesfully sent

I also tried adding a new function with the process running, but it didn't work until after restarting the process:
1. Create new function definition in `manifest.ts` pointing at a file that doesn't exist
2. Add new function to the functions array
3. Create the new function file
4. ---> Shortcut doesn't appear in Slack until after restarting the process

**Comment by filmaj** [comment]
Oooo good catch @shapirone ! Totally slipped my mind. This worked in the old deno sdk, ya? This is what the file-watching stuff was in place for IIRC.... I will take a look tomorrow!

**Comment by filmaj** [comment]
@shapirone OK I think I fixed that up. I needed to add the filewatching paths specifier to the slack.json to instruct the CLI when to re-install the app to the workspace. The only time that would be needed is when the manifest changes. Runtime code for the app doesn't need to trigger workspace re-installs because functions get loaded dynamically in a fresh process with every event payload incoming from the Slack backend.

File: .slack/slack.json [hunk 1]
```diff
@@ -8,5 +8,14 @@
...
       "default": "deno run -q --unstable --import-map=import_map.json --allow-read --allow-write --allow-net https://deno.land/x/deno_slack_builder@0.0.5/mod.ts"
     }
+  },
+  "run": {
+    "script": {
+      "default": "deno run -q --unstable --allow-write --allow-read --allow-net https://deno.land/x/deno_slack_runtime@0.0.2/mod.ts"
+    },
+    "watcher": {
+      "filter_regex": "^.*\\.ts$",
+      "paths": [ "./manifest.ts" ]
```
Comment (mwbrooks) [inline .slack/slack.json]: ~Do we want to support all supported types like the builder?~ Hmm.... the watcher freaks out when the files are missing, so this would need to be solved with a regex.
```
      "paths": [ "./manifest.json", "./manifest.js", "./manifest.ts" ]
```

**Review by mwbrooks — COMMENTED** [review]
I haven't taken the time to re-create @shapirone's scenario, but it got me thinking that we may want to support the flavours of the manifest file too. When any changes, we'll need to update your manifest.

File: .slack/slack.json (hunk 1 above)
Comment (shapirone) [reply .slack/slack.json]: ➕ ➕ ➕  but I'm not sure if this also requires an update to the `filter_regex`

I think this should be the standard provided via `get-hooks` as well

Comment (filmaj) [reply .slack/slack.json]: For deno / run-on-slack projects, we can just drop the regex altogether (based on how the [slack-cli handles this property](https://github.com/slackapi/slack-cli/blob/main/pkg/cloud/run.go#L242-L245) - it is ignored if it is missing) since we can, as @mwbrooks suggests, just explicitly reference all three possible manifest filetypes. In these projects, since each function-execution event coming from the Slack backend causes the CLI to spin up a new `deno` process wrapping the user function execution (the short-lived process model, as opposed to Bolt, which uses a long-lived process model), we only need to worry about changes to the manifest files here.

**Comment by filmaj** [comment]
@mwbrooks @shapirone I think I have a solution in place now in this PR to watch for changes to any of the potential manifest files and re-install the app appropriately, without those annoying 'file does not exist' golang errors raised by the CLI I mentioned in Slack earlier.

**Review by mwbrooks — APPROVED** [review]
✅ Local run work
✅ Updates to `manifest.ts` works
